---
graphs:
  
  - type: pos_vel_eff
    lines: [Position (rad), Velocity (rad/s), Effort]
    ranges: [[-3.14, 3.14], [-3.14, 3.14], [-600, 600]]
    graph_names: [thj1, thj2, thj3, thj4, thj5, ffj1, ffj2, ffj3, ffj4, mfj1, mfj2, mfj3, mfj4, rfj1, rfj2, rfj3, rfj4, lfj1, lfj2, lfj3, lfj4, lfj5, wrj1, wrj2]
    colours: [red, blue, green]
    font_size: 6
    topic_namespace: /joint_states
    message_type: JointState

  - type: control_loops
    lines: [Setpoint (rad), Input (rad), dInput/dt (rad/s), Error, Output]
    ranges: [[-4, 4], [-4, 4], [-4, 4], [-4, 4], [-600, 600]]
    graph_names: [ffj0, ffj3, ffj4, mfj0, mfj3, mfj4, thj1, thj2, thj3, thj4, thj5, rfj0, rfj3, rfj4, lfj0, lfj3, lfj4, lfj5, wrj1, wrj2]
    colours: [red, blue, green, purple, cyan]
    font_size: 6
    legend_columns: 3
    topic_namespace_start: /sh_lh_
    topic_namespace_end: _position_controller/state
    message_type: JointControllerState

  - type: motor_stat
    lines: ['Strain Gauge Left', 'Strain Gauge Right', 'Measured PWM', 'Measured Current (A)', 'Measured Voltage (V)', 'Measured Effort', 'Temperature (ÂºC)', 'Unfiltered position', 'Unfiltered force', 'Last Commanded Effort', 'Encoder Position (rad)']
    ranges: [[-550, 550], [-550, 550], [-300, 300], [0, 0.8], [-40, 40], [-1000, 1000], [0, 35], [-300, 300], [-450, 450], [-600, 600], [-6, 6]]  
    graph_names: [ffj0, ffj3, ffj4, mfj0, mfj3, mfj4, thj1, thj2, thj3, thj4, thj5, rfj0, rfj3, rfj4, lfj0, lfj3, lfj4, lfj5, wrj1, wrj2]
    colours: ['red', 'cyan', 'green', 'purple', 'yellow', 'blue', 'green', 'pink', 'red', 'magenta', 'red']
    font_size: 8
    topic_namespace: /diagnostics

  - type: palm_extras_accelerometer
    lines: ['Accel X', 'Accel Y', 'Accel Z']
    ranges: [[-200, 200], [-200, 200], [-200, 200]]
    graph_names: [palm_extras_accelerometer]
    colours: ['red', 'cyan', 'green']
    font_size: 12
    topic_namespace: /lh/palm_extras

  - type: palm_extras_gyro
    lines: ['Gyro x', 'Gyro y', 'Gyro z']
    ranges: [[-200, 200], [-200, 200], [-200, 200]]
    graph_names: [palm_extras_gyro]
    colours: ['purple', 'yellow', 'blue']
    font_size: 12
    topic_namespace: /lh/palm_extras

  - type: palm_extras_adc
    lines: ['ADC0', 'ADC1', 'ADC2', 'ADC3']
    ranges: [[0, 50], [0, 50], [0, 50], [0, 50]]
    graph_names: [palm_extras_adc]
    colours: ['red', 'cyan', 'green', 'purple']
    font_size: 12
    topic_namespace: /lh/palm_extras

  - type: biotacs
    lines: ['biotac_0', 'biotac_1', 'biotac_2', 'biotac_3', 'biotac_4']
    ranges: [[2000, 2200], [1000, 2000], [0, 5000], [0, 5000], [750, 1500]]
    graph_names: ['TAC', 'TDC', 'PAC0', 'PAC1', 'PDC']
    colours: ['red', 'cyan', 'green', 'purple', 'blue']
    font_size: 10
    topic_namespace: /lh/tactile
